### PR TITLE
Add webhook for yanked versions

### PR DESCRIPTION
**Pull request by greysteil**
Rubygems' webhooks are awesome. We use them to trigger dependency updates with Dependabot.

Sometimes, gems get yanked. Currently, rubygems doesn't send a webhook when this happens.

This PR changes that - it adds a `yanked` attribute to the payload serialized in webhooks for a given version, and adds a trigger to send webhooks when a gem is yanked. The webhook sent is the same as the one sent when a new version is pushed (except the new yanked attribute will be true).

I'd like to use this at Dependabot to automatically close any PRs that we've open to yanked versions, and downgrade people if necessary.

### CODE REVIEW

**Comment by dwradcliffe** [comment]
I haven’t reviewed the code (yet) but I like the idea!

**Comment by greysteil** [comment]
@dwradcliffe glad you're 👍 on the idea! Code is pretty straightforward, and we shouldn't be firing too many of these so hopefully no ops issues.

**Comment by feelepxyz** [comment]
@dwradcliffe any chance to get this reviewed? 🙏

File: app/controllers/api/v1/deletions_controller.rb [hunk 1]
```diff
@@ -46,4 +47,16 @@ def validate_gem_and_version
       end
     end
   end
+
+  def enqueue_web_hook_jobs
```
Comment (sonalkr132) [inline app/controllers/api/v1/deletions_controller.rb]: Can we please move this and one used in Pusher to application controller? Call `enqueue_web_hook_jobs` after `gemcutter.process`.
https://github.com/rubygems/rubygems.org/blob/00468dc5fb20f6ae27c3d53528ed654b13cded51/app/models/pusher.rb#L126

File: app/models/rubygem.rb [hunk 2]
```diff
@@ -153,6 +153,7 @@ def payload(version = versions.most_recent, protocol = Gemcutter::PROTOCOL, host
       'info'              => version.info,
       'licenses'          => version.licenses,
       'metadata'          => version.metadata,
+      'yanked'            => version.yanked?,
```
Comment (sonalkr132) [inline app/models/rubygem.rb]: Please explain rationale of adding this in the commit message.

**Comment by greysteil** [comment]
@sonalkr132 - updated! Sorry for the delay on this one.

File: app/models/rubygem.rb (hunk 2 above)
Comment (greysteil) [reply app/models/rubygem.rb]: Done!

**Comment by sonalkr132** [comment]
Thanks @greysteil . Had to make one more change. Check: https://github.com/rubygems/rubygems.org/pull/1934

**Comment by greysteil** [comment]
Awesome, thanks @sonalkr132!

**Pull request closed without merging**